ebooks.py

```diff
@@ -34,119 +38,143 @@ def entity(text):
         try:
             text = chr(numero)
         except KeyError:
-            pass    
+            pass
     return text
 
+
 def filter_tweet(tweet):
-    tweet.text = re.sub(r'\b(RT|MT) .+','',tweet.text) #take out anything after RT or MT
-    tweet.text = re.sub(r'(\#|@|(h\/t)|(http))\S+','',tweet.text) #Take out URLs, hashtags, hts, etc.
-    tweet.text = re.sub(r'\n','', tweet.text) #take out new lines.
-    tweet.text = re.sub(r'\"|\(|\)', '', tweet.text) #take out quotes.
-    tweet.text = re.sub(r'\s+\(?(via|says)\s@\w+\)?', '', tweet.text) # remove attribution
+    tweet.text = re.sub(r'\b(RT|MT) .+', '', tweet.text)  # take out anything after RT or MT
+    tweet.text = re.sub(r'(\#|@|(h\/t)|(http))\S+', '', tweet.text)  # Take out URLs, hashtags, hts, etc.
+    tweet.text = tweet.text.replace('\n', '')  # take out new lines.
+    tweet.text = re.sub(r'\"|\(|\)', '', tweet.text)  # take out quotes.
+    tweet.text = re.sub(r'\s+\(?(via|says)\s@\w+\)?', '', tweet.text)  # remove attribution
     htmlsents = re.findall(r'&\w+;', tweet.text)
-    if len(htmlsents) > 0 :
-        for item in htmlsents:
-            tweet.text = re.sub(item, entity(item), tweet.text)    
-    tweet.text = re.sub(r'\xe9', 'e', tweet.text) #take out accented e
+    for item in htmlsents:
+        tweet.text = tweet.text.replace(item, entity(item))
+    tweet.text = tweet.text.replace('\xe9', 'e')  # take out accented e
     return tweet.text
-                     
-                     
-                                                    
+
+
+def scrape_page(src_url, web_context, web_attributes):
+    tweets = []
+    last_url = ""
+    for i in range(len(src_url)):
+        if src_url[i] != last_url:
+            last_url = src_url[i]
+            print(">>> Scraping {0}".format(src_url[i]))
+            try:
+                page = urlopen(src_url[i])
+            except Exception:
+                last_url = "ERROR"
+                import traceback
+                print(">>> Error scraping {0}:".format(src_url[i]))
+                print(traceback.format_exc())
+                continue
+            soup = BeautifulSoup(page, 'html.parser')
+        hits = soup.find_all(web_context[i], attrs=web_attributes[i])
+        for hit in hits:
+            tweet = str(hit.text).strip()
```

Comment:
This returns a UnicodeEncodeError in Python2. So there should be an if statement here to catch the exception.

Reply:
Handled! I think the `UnicodeDecodeError` might be superfluous. Let me know if you'd like me to remove.